credentialproviderpackage/Makefile

```diff
@@ -0,0 +1,41 @@
+SHELL = /usr/bin/env bash -o pipefail
+.SHELLFLAGS = -ec
+
+REPO_ROOT=$(shell git rev-parse --show-toplevel)/credentialproviderpackage
+GOLANG_VERSION?="1.18"
+#GO ?= $(shell source $(REPO_ROOT)/scripts/common.sh && build::common::get_go_path $(GOLANG_VERSION))/go
+GO = go
+
+# Get the currently used golang install path (in GOPATH/bin, unless GOBIN is set)
+ifeq (,$(shell go env GOBIN))
+GOBIN=$(shell go env GOPATH)/bin
+else
+GOBIN=$(shell go env GOBIN)
+endif
+
+all: build
+
+help: ## Display this help.
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z_0-9-]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
+
+clean: ## Clean output directory, and the built binary
+	rm -rf output/
+	rm -rf bin/*
+	rm cover.out
+
+##@ Build
+
+build: ## Build Binary
+	mkdir -p $(REPO_ROOT)/bin
+	$(GO) mod tidy -compat=$(GOLANG_VERSION)
+	$(GO) build -o $(REPO_ROOT)/bin/aws-credential-provider $(REPO_ROOT)/cmd/aws-credential-provider/*.go
+
+build-linux:
+	[ -d bin ] || mkdir bin
+	env CGO_ENABLED=0 GOOS=linux GOARCH=amd64 $(MAKE) build
```

Comment:
Don't we need to build this for linux arm and amd?

Reply:
I didn't think we supported ARM for the actual cluster but I can add it. Should work the same on ARM

Reply:
We're not going to use this makefile for the actual image build right? This will be in build-tooling.

Reply:
The package controller does do arch or os builds
```
build: go.sum generate ## Build package-manager binary.
    $(GO) build -o $(BIN_DIR)/package-manager main.go
```